Optional/src/test/java/by/solbegsoft/oprional/impl/OptionalMethodImplTest.java

```diff
@@ -7,15 +7,60 @@
 
 import java.util.Arrays;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class OptionalMethodImplTest {
 
+    private final OptionalMethod optionalMethod = new OptionalMethodImpl();
+
     @Test
     void userExecutor() {
-        OptionalMethod optionalMethod = new OptionalMethodImpl();
-        String expected = "DEFAULT,12,reallycomplexmail222@hh.ru";
-        String actual = optionalMethod.userExecutor(new User(null, 12, Arrays.asList("reallycomplexmail222@hh.ru")));
+
+        User user = new User(null, 12, Arrays.asList("reallycomplexmail222@hh.ru"));
+        String expected = user.toString();
+        String actual = optionalMethod.userExecutor(user);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void userExecutorNull() {
+        User user = null;
+        String expected = new RuntimeException("nullable user").getMessage();
+        RuntimeException actual = assertThrows(
+                RuntimeException.class,
+                () -> optionalMethod.userExecutor(user));
+        Assertions.assertEquals(expected, actual.getMessage());
+    }
+
+    @Test
+    void usernameExecutor() {
+        User user = new User("Kate", 12, Arrays.asList("reallycomplexmail222@hh.ru"));
+        String expected = user.getName();
+        String actual = optionalMethod.usernameExecutor(user);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void usernameExecutorNull() {
+        User user = new User(null, 12, Arrays.asList("reallycomplexmail222@hh.ru"));
+        String expected = "DEFAULT";
+        String actual = optionalMethod.usernameExecutor(user);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void userEmailExecutor() {
+        User user = new User("Ivan", 14, Arrays.asList("idontliketorepeattwice@mail.ru", "idontliketorepeattwice@mail.ru", "msin@fds.com"));
+        String expected = String.join(",", user.getEmails());
+        String actual = optionalMethod.userEmailExecutor(user);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void userEmailExecutorNull() {
```

Comment:
It will be in another module, but try to name test's methods more conscious 
FE
`testExecuteUserWithNullEmail() `